bazel/repository_locations.bzl

```diff
@@ -14,42 +14,47 @@ REPOSITORY_LOCATIONS = dict(
         strip_prefix = "thrift-0.11.0",
     ),
     com_github_bombela_backward = dict(
-        commit = "44ae9609e860e3428cd057f7052e505b4819eb84",  # 2018-02-06
+        sha256 = "ad73be31c5cfcbffbde7d34dba18158a42043a109e7f41946f0b0abd589ed55e",
+        commit = "v1.4",
```

Comment:
I'm not sure exactly what this is trying to express, but it won't do what you want.

1. Setting a SHA256 can only work when downloading a single file, such as a tarball. Git clones don't have a deterministic sha256 so there's nowhere the `sha256` value could be used or validated.

2. The `commit` kwarg to `git_repository` is expected to be a Git commit ID. For fetching by tag, you want the `tag` kwarg. Note that this is usually considered less secure + deterministic than fetching by commit, because Git tags can be arbitrarily changed by the repository owner.

Reply:
I think tag is nicer if we can get a SHA256 for verification; we get self-documenting specs and a stronger hash than that provided by native git.

Reply:
What byte blob is the sha256 calculated from?

Reply:
See prior comments https://github.com/envoyproxy/envoy/pull/4353#issuecomment-419088327 and  https://github.com/envoyproxy/envoy/pull/4353/commits/fe04e3913f8294931c41b78504608e302691748f#r215618035. kwarg of `tag` generates an error. By specifying the sha256 we are downloading the tar file.

Reply:
Are you sure? Envoy's repositories.bzl uses the Skylark implementation of `git_repository` from https://github.com/bazelbuild/bazel/blob/master/tools/build_defs/repo/git.bzl, which doesn't have any hits for "sha256" or "github".

Reply:
I think @jmillikin-stripe is right. Adding an invalid sha passes through without error in `bazel/repository_locations.bzl`. Passing in `fakevalue` instead of `sha256` works too so Bazel is ignoring what it doesn't comprehend. 

My guess is that https://github.com/envoyproxy/envoy/blob/master/bazel/repository_locations.bzl#L33-L35 works as it invokes https://github.com/bazelbuild/bazel/blob/master/tools/build_defs/repo/http.bzl which does support `sha256` - https://github.com/bazelbuild/bazel/blob/master/tools/build_defs/repo/http.bzl#L122

In summary, using git means we can't check the sha. We would need to move away from tags to commit ID AND use the http method. 

@htuch the original vision of greater maintainability with sha256 checked security isn't panning out :-(

Reply:
If you're willing to have a stricter dependency on Github and a slightly risk of build breakage when they change their tarball builder, you could use Github's auto-generated tarballs with sha256 attrs.

Reply:
@jmillikin-stripe do you think GH have reduced their churn? I remember the original concern due to GH breaking arbitrarily URLs and SHAs, has this stabilized since? If so, that seems entirely desirable.